File: src/NeanderSharp/NeanderSharp.Tests/CalculatorTests.cs
using System;
using System.Text;
using System.Collections.Generic;

using Microsoft.VisualStudio.TestTools.UnitTesting;
using NeanderSharp;

namespace TestsNeander
{
    /// <summary>
    /// Summary description for CalculetorTests
    /// </summary>
    [TestClass]
    public class CalculatorTests
    {
        string memfile;
        Neander n;
        public CalculatorTests()
        {
            //memfile = @"C:\Users\Filipe\Documents\git\Simple-C-Compiler\mem\neander\calculadora-lfzawacki.mem";
            //memfile = @"C:\Users\Filipe\Documents\git\Simple-C-Compiler\mem\neander\calculadora-bjurkowski.MEM";
            //memfile = @"C:\Users\Filipe\Documents\git\Simple-C-Compiler\mem\neander\calculadora-cmdalbem.MEM";
            memfile = @"C:\Users\Filipe\Documents\git\Simple-C-Compiler\mem\neander\calculadora-labianchin.mem";
            n = new Neander();
            ReadMemToNeander(memfile, n);
        }

        private void ReadMemToNeander(string file, Neander n)
        {
            MemReader r = new MemReader(file);
            r.ReadHeader();
            for (int i = 0; i <= 255; i++)
            {
                n.Memory[i] = r.ReadByte();
            }
            r.Close();
        }

        [TestMethod]
        public void Case1()
        {
            //caso 1
            n.Reset();
            n.Memory[128] = 1;
            n.Memory[129] = 128;
            n.Memory[130] = 8;
            n.Run();
            Assert.AreEqual(8, n.Memory[131]);
        }


        [TestMethod]
        public void Case2()
        {
            //caso 2
            n.Reset();
            n.Memory[128] = 3;
            n.Memory[129] = 44;
            n.Memory[130] = 1;
            n.Run();
            Assert.AreEqual(43, n.Memory[131]);
        }

        [TestMethod]
        public void Case3()
        {
            //caso 3
            n.Reset();
            n.Memory[128] = 1;
            n.Memory[129] = 29;
            n.Memory[130] = 1;
            n.Run();
            Assert.AreEqual(30, n.Memory[131]);
        }

        [TestMethod]
        public void Case4()
        {
            //caso 4
            n.Reset();
            n.Memory[128] = 15;
            n.Memory[129] = 115;
            n.Memory[130] = 15;
            n.Run();
            Assert.AreEqual(57, n.Memory[131]);
        }
        [TestMethod]
        public void Case5()
        {
            //caso 5
            n.Reset();
            n.Memory[128] = 7;
            n.Memory[129] = 105;
            n.Memory[130] = 127;
            n.Run();
            Assert.AreEqual(210, n.Memory[131]);
        }

        [TestMethod]
        public void Case6()
        {
            //caso 6
            n.Reset();
            n.Memory[128] = 3;
            n.Memory[129] = 47;
            n.Memory[130] = 3;
            n.Run();
            Assert.AreEqual(44, n.Memory[131]);
        }

        [TestMethod]
        public void Case7()
        {
            //caso 7
            n.Reset();
            n.Memory[128] = 7;
            n.Memory[129] = 122;
            n.Memory[130] = 100;
            n.Run();
            Assert.AreEqual(244, n.Memory[131]);
        }

        [TestMethod]
        public void Case8()
        {
            //caso 8
            n.Reset();
            n.Memory[128] = 15;
            n.Memory[129] = 28;
            n.Memory[130] = 0;
            n.Run();
            Assert.AreEqual(14, n.Memory[131]);
        }

        [TestMethod]
        public void Case9()
        {
            //caso 9
            n.Reset();
            n.Memory[128] = 1;
            n.Memory[129] = 156;
            n.Memory[130] = 8;
            n.Run();
            Assert.AreEqual(20, n.Memory[131]);
        }

        [TestMethod]
        public void Case10()
        {
            //caso 10
            n.Reset();
            n.Memory[128] = 3;
            n.Memory[129] = 144;
            n.Memory[130] = 1;
            n.Run();
            Assert.AreEqual(15, n.Memory[131]);
        }

        [TestMethod]
        public void Case11()
        {
            //caso 11
            n.Reset();
            n.Memory[128] = 1;
            n.Memory[129] = 129;
            n.Memory[130] = 1;
            n.Run();
            Assert.AreEqual(0, n.Memory[131]);
        }

        [TestMethod]
        public void Case12()
        {
            //caso 8
            n.Reset();
            n.Memory[128] = 15;
            n.Memory[129] = 130;
            n.Memory[130] = 1;
            n.Run();
            Assert.AreEqual(1, n.Memory[131]);
        }

        [TestMethod]
        public void Case13()
        {
            //caso 13
            n.Reset();
            n.Memory[128] = 7;
            n.Memory[129] = 0;
            n.Memory[130] = 127;
            n.Run();
            Assert.AreEqual(0, n.Memory[131]);
        }

        [TestMethod]
        public void Case14()
        {
            //caso 14
            n.Reset();
            n.Memory[128] = 3;
            n.Memory[129] = 130;
            n.Memory[130] = 131;
            n.Run();
            Assert.AreEqual(1, n.Memory[131]);
        }

        [TestMethod]
        public void Case15()
        {
            //caso 15
            n.Reset();
            n.Memory[128] = 7;
            n.Memory[129] = 122;
            n.Memory[130] = 100;
            n.Run();
            Assert.AreEqual(244, n.Memory[131]);
        }

        [TestMethod]
        public void Case16()
        {
            //caso 16
            n.Reset();
            n.Memory[128] = 15;
            n.Memory[129] = 127;
            n.Memory[130] = 0;
            n.Run();
            Assert.AreEqual(63, n.Memory[131]);
        }

    }
}

File: src/PyNeander/calculatortests.py
from neander import *
import unittest

class CalculatorTestNeander(unittest.TestCase):

	def setUp(self):
		self.neander = Neander()
		#self.neander.from_file('/home/kidbomb/git/simple-c-compiler/mem/neander/calculadora-bjurkowski.mem')#7
		#self.neander.from_file('/home/kidbomb/git/simple-c-compiler/mem/neander/calculadora-lfzawacki.mem')#7
		#self.neander.from_file('/home/kidbomb/git/simple-c-compiler/mem/neander/calculadora-cmdalbem.mem')#OK
		self.neander.from_file('/home/kidbomb/git/simple-c-compiler/mem/neander/calculadora-labianchin.mem')#OK

	def test_case1(self):
		self.neander.reset()
		self.neander.memory[128] = 1
		self.neander.memory[129] = 128
		self.neander.memory[130] = 8
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 8)

	def test_case2(self):
		self.neander.reset()
		self.neander.memory[128] = 3
		self.neander.memory[129] = 44
		self.neander.memory[130] = 1
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 43)

	def test_case3(self):
		self.neander.reset()
		self.neander.memory[128] = 1
		self.neander.memory[129] = 29
		self.neander.memory[130] = 1
		print "Start run"
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 30)

	def test_case4(self):
		self.neander.reset()
		self.neander.memory[128] = 15
		self.neander.memory[129] = 115
		self.neander.memory[130] = 15
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 57)

	def test_case5(self):
		self.neander.reset()
		self.neander.memory[128] = 7
		self.neander.memory[129] = 105
		self.neander.memory[130] = 127
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 210)

	def test_case6(self):
		self.neander.reset()
		self.neander.memory[128] = 3
		self.neander.memory[129] = 47
		self.neander.memory[130] = 3
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 44)

	def test_case7(self):
		self.neander.reset()
		self.neander.memory[128] = 7
		self.neander.memory[129] = 122
		self.neander.memory[130] = 100
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 244)

	def test_case8(self):
		self.neander.reset()
		self.neander.memory[128] = 15
		self.neander.memory[129] = 28
		self.neander.memory[130] = 0
		self.neander.run()

		self.assertEqual(self.neander.memory[131], 14)

if __name__ == '__main__':
	unittest.main()
